src/Storage/Controllers/InstancesController.cs

```diff
@@ -183,7 +184,30 @@ public class InstancesController : ControllerBase
                     return BadRequest("Invalid InstanceOwnerIdentifier.");
                 }
 
-                (string person, string orgNo) = InstanceHelper.SeparatePersonAndOrgNo(instanceOwnerIdType, instanceOwnerIdValue);
+                string person = null;
+                string orgNo = null;
+
+                if (Enum.TryParse<PartyType>(instanceOwnerIdType, true, out PartyType partyType))
+                {
+                    if (partyType == PartyType.Person)
+                    {
+                        if (instanceOwnerIdValue.Length != 11)
```

Comment:
We might want to add a regex here. To ensure that it's 11 digits and not a mix off digits and letters. The current check doesn't catch `hhhhhhhhhhh` for example. It would fall through and we would have an unnecessary call to Register.